src/query/typed_value.cpp

```diff
@@ -362,6 +362,33 @@ bool TypedValue::IsGraph() const { return type_ == Type::Graph; }
 
 #undef DEFINE_VALUE_AND_TYPE_GETTERS
 
+bool TypedValue::ContainsDeleted() const {
+  switch (type_) {
+    case Type::List:
+      for (const auto &elem : list_v) {
+        if (elem.ContainsDeleted()) return true;
+      }
+      return false;
+    case Type::Map:
+      for (const auto &[_, map_value] : map_v) {
+        if (map_value.ContainsDeleted()) return true;
+      }
+      return false;
+    case Type::Vertex:
+      return vertex_v.impl_.vertex_->deleted;
+    case Type::Edge:
+      return edge_v.impl_.edge_.ptr->deleted;
+    case Type::Path:
+      for (const auto &vertex_acc : path_v.vertices()) {
+        if (vertex_acc.impl_.vertex_->deleted) return true;
+      }
+      std::ranges::any_of(path_v.edges(), [](auto &edge_acc) { return edge_acc.impl_.edge_.ptr->deleted; });
```

Comment:
I think we should add `return` before `std::ranges::any_of...`

Reply:
Done